components/dashboard/src/App.test.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { getURLHash } from './App'
+
+test('urlHash', () => {
+    global.window = Object.create(window);
+    Object.defineProperty(window, 'location', {
+        value: {
+            href: '#http://example.org/',
+            hash: '#examples'
+        }
```

Comment:
Many thanks for adding tests! 💯 👏

I'm just slightly confused as to what problem this test tries to address. In which situation can your `window.location` have both a `href` with `#` _and_ a `hash`? 🤔 Is this related to a bug we've seen? (Maybe a small comment could help clarify this for future readers 😇)

Reply:
Lol I completely missed this. Initially I wanted to add this simple test because my Regex skills is pretty non-existent (and we have a bunch in App.tsx), so I wanted to continue playing around with tests but ended up forgetting about it and focusing on the feature. Will fix this, and if we do a sync review maybe we can try more tests together.

Reply:
Oh I misread the question, I thought you meant my `href` was wrong (which it is) but `href` and `hash` can both be there. The `href` is basically the same as `window.location`, and the `hash` is only the part starting from the `#`.